infinispan/src/main/java/org/hibernate/search/infinispan/impl/InfinispanDirectoryProvider.java

```diff
@@ -129,8 +136,30 @@ public void start(DirectoryBasedIndexManager indexManager) {
 		log.debugf( "Initialized Infinispan index: '%s'", directoryProviderName );
 	}
 
+	private void augmentDirectoryWithAsyncDeletes(org.infinispan.lucene.directory.BuildContext directoryBuildContext) {
+		Executor executor = deletesExecutor.getExecutor();
+		try {
+			Method operationsExecutor = directoryBuildContext.getClass().getMethod(
+					"deleteOperationsExecutor",
+					Executor.class
+			);
+			operationsExecutor.invoke( directoryBuildContext, executor );
+		}
+		catch (NoSuchMethodException e) {
+			log.notSettingAsyncDeletes();
+		}
+		catch (InvocationTargetException e) {
+			log.notSettingAsyncDeletes();
```

Comment:
I like your point of not scaring users, +1 to not add the Cause clause.
But consider also what you would need in case you were to support someone to try figure out why the performance didn't improve as expected? I'd use at least a slightly different message to understand if
 # It loaded the wrong Infinispan version 
 # Some other (security?) reason prevented you to perform the reflective call